engine/src/main/java/org/entando/entando/web/user/validator/UserValidator.java

```diff
@@ -209,8 +209,15 @@ private void checkNewPassword(String username, String password, BindingResult bi
     }
 
     private boolean verifyPassword(String username, String password) {
-        UserDetails user = this.extractUser(username);
-        return passwordEncoder.matches(password, user.getPassword());
+        try {
+            UserDetails user = this.getUserManager().getUser(username, password);
```

Comment:
Does this call throw an exception if the password doesn't match?

Reply:
Yes. It first sets:
` bindingResult.rejectValue("oldPassword", ERRCODE_OLD_PASSWORD_FORMAT, new String[]{}, "user.password.old.invalid");`
Then:
`throw new ValidationGenericException(bindingResult);`